tests/positive/proxy/http.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/coreos/ignition/tests/register"
+	"github.com/coreos/ignition/tests/types"
+)
+
+var (
+	proxyServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(ignConfig))
+	}))
+
+	// configServer httptest.Server
+	configServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(ignConfig))
+	}))
+
+	mockConfigURL = "http://www.fake.tld"
+
+	ignConfig = `{
+		"ignition": { "version": "3.0.0-experimental" },
+		"storage": {
+			"files": [{
+				"path": "/foo/bar",
+				"contents": { "source": "data:,example%20file%0A" }
+			}]
+		}
+	}`
+)
+
+func init() {
+	register.Register(register.PositiveTest, CanUseProxyForRetrievingConfig())
+	register.Register(register.PositiveTest, CanUseNoProxyForRetrievingConfig())
+}
+
+func CanUseProxyForRetrievingConfig() types.Test {
+	name := "Can Use Proxy For Retrieving Config"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := fmt.Sprintf(`{
+		"ignition": {
+			"version": "$version",
+			"config": {
+				"replace": {
+					"source": "%s/"
+				}
+			},
+			"proxy": {
+				"httpProxy": "%s",
+				"httpsProxy": "%s",
+				"noProxy": ["localhost", "127.0.0.1"]
+			}
+		}
+	}`, mockConfigURL, proxyServer.URL, proxyServer.URL)
+	configMinVersion := "3.0.0-experimental"
+
+	return types.Test{
+		Name:             name,
+		In:               in,
+		Out:              out,
+		Config:           config,
+		ConfigMinVersion: configMinVersion,
+	}
+}
+
+func CanUseNoProxyForRetrievingConfig() types.Test {
+	name := "Can Use No Proxy For Retrieving Config"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := fmt.Sprintf(`{
+		"ignition": {
+			"version": "$version",
+			"config": {
+				"replace": {
+					"source": "%s/"
+				}
+			},
+			"proxy": {
+				"httpProxy": "%s",
+				"httpsProxy": "%s",
+				"noProxy": ["localhost", "127.0.0.1"]
+			}
+		}
+	}`, configServer.URL, proxyServer.URL, proxyServer.URL)
+	configMinVersion := "3.0.0-experimental"
+
```

Comment:
Should verify the other parts of the config were applied. Also serve a different config so we you that `noProxy` took effect.

Reply:
Change addressed in latest rebase. I decided to remove the configServer from the test file and re-use the one in servers.go to verify work as expected. Let me know what you think?